lib/MusicBrainz/Server/Controller/Account.pm

```diff
@@ -705,6 +705,18 @@ sub register_application : Path('/account/applications/register') RequireAuth Re
         });
         $c->response->redirect($c->uri_for_action('/account/applications'));
         $c->detach;
+    } else {
+        $form->field('oauth_redirect_uri')->value('');
+        $form->field('oauth_type')->value('web');
+        $form->field('name')->value('');
```

Comment:
These will clear/change the fields if there's an error in the submission, though.

After I remove these lines, I can't seem to reproduce the warning you're getting in https://github.com/metabrainz/musicbrainz-server/pull/708#discussion_r212809463 (which I assumed this is meant to resolve). How can I trigger that?

Reply:
Thanks! Now I see the horrible error.

I think I agree with your last sentence (setting the `default` in the form class), so that the component doesn't have to deal with weird data. There doesn't seem to be any purpose in keeping the default value to `undef` here (if anywhere; not sure why FormHandler does that, but perhaps to distinguish between unset fields and fields explicitly set to `''`). At least now we know about the issue.

Reply:
Applied with https://github.com/metabrainz/musicbrainz-server/pull/708/commits/c76f6eb4461f9b03162201d60689bb6b7338b8ad.

Note: In either case, unset fields should be set to JS `undefined` not JS `null` for React.